### PR TITLE
Address Adam's comments

### PR DESCRIPTION
**Pull request by karya0**
* Added authentication.service_account_secret_name.

### CODE REVIEW

File: repo/packages/S/spark/12/config.json [hunk 1]
```diff
@@ -129,10 +129,10 @@
                             "description": "Mesos scheduler dcos-oauth configuration.",
                             "type": "object",
                             "properties": {
-                                "enabled": {
-                                    "description": "Whether to enable dcos-oauth.",
-                                    "type": "boolean",
-                                    "default": false
+                                "service_account_secret_name": {
```
Comment (adam-mesos) [inline repo/packages/S/spark/12/config.json]: Is this what hdfs calls it? It'd be nice to be consistent between the two if possible.

Comment (karya0) [inline repo/packages/S/spark/12/config.json]: Yes, @nickbp and I agreed on this name yesterday.

**Comment by adam-mesos** [comment]
:rocket:

File: repo/packages/S/spark/12/config.json [hunk 2]
```diff
@@ -129,10 +129,10 @@
...
+                                    "default": "",
```
Comment (mgummelt) [inline repo/packages/S/spark/12/config.json]: Let's remove this default.  An empty string is not the same as unset.

File: repo/packages/S/spark/12/config.json (hunk 1 above)
Comment (mgummelt) [inline repo/packages/S/spark/12/config.json]: Let's call this "secret_name".  It's implicit that it's for the service.

**Pull request merged**